src/core/geometry/qgsgeometry.h

```diff
@@ -630,15 +630,15 @@ class CORE_EXPORT QgsGeometry
      * Returns the vertex closest to the given point, the corresponding vertex index, squared distance snap point / target point
      * and the indices of the vertices before and after the closest vertex.
      * \param point point to search for
-     * \param atVertex will be set to the vertex index of the closest found vertex
-     * \param beforeVertex will be set to the vertex index of the previous vertex from the closest one. Will be set to -1 if
+     * \param closestVertexIndex will be set to the vertex index of the closest found vertex
+     * \param previousVertexIndex will be set to the vertex index of the previous vertex from the closest one. Will be set to -1 if
      * not present.
-     * \param afterVertex will be set to the vertex index of the next vertex after the closest one. Will be set to -1 if
+     * \param nextVertexIndex will be set to the vertex index of the next vertex after the closest one. Will be set to -1 if
      * not present.
      * \param sqrDist will be set to the square distance between the closest vertex and the specified point
      * \returns closest point in geometry. If not found (empty geometry), returns null point and sqrDist is negative.
      */
-    QgsPointXY closestVertex( const QgsPointXY &point, int &atVertex SIP_OUT, int &beforeVertex SIP_OUT, int &afterVertex SIP_OUT, double &sqrDist SIP_OUT ) const;
+    QgsPointXY closestVertex( const QgsPointXY &point, int &closestVertexIndex SIP_OUT, int &previousVertexIndex SIP_OUT, int &nextVertexIndex SIP_OUT, double &sqrDist SIP_OUT ) const;
```

Comment:
There's a TODO note a couple of lines above the documentation here which you can safely remove now